File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let map=new Map();
    for (let num of nums){
        map.set(num,(map.get(num)||0)+1);
    }
    let freq=0;
    let ans=nums[0];
    for(let a of map.keys()){
        if(map.get(a)>freq){
            freq=map.get(a);
            ans=a;
        }
    }
    return ans;
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cand=0,count=0;
        for(int a:nums){
            if(count==0){
                cand=a;
            }
            if(cand==a){
                count++;
            }else {
                count--;
            }
        }
        return cand;
    }
}